test/es-module/test-esm-loader-hooks.mjs

```diff
@@ -747,6 +747,25 @@ describe('Loader hooks', { concurrency: true }, () => {
     assert.strictEqual(signal, null);
   });
 
+  it('should support source maps in commonjs translator', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--enable-source-maps',
+      '--import',
+      `data:text/javascript,${encodeURIComponent(
+        'import{ register } from "node:module";' +
+        'import { pathToFileURL } from "node:url";' +
+        'register("./hooks-commonjs.mjs", pathToFileURL("./"));'
```

Comment:
Sorry for being difficult, but I think we should keep the hook in the test file, as using an external file makes it harder to follow IMO
```suggestion
        'register(${'`'}data:text/javascript,import { readFile } from 'node:fs/promises';export ${'${'}encodeURIComponenent(${
         async function load(url, context, nextLoad) {
            const resolved = await nextLoad(url, context);
            if (context.format === 'commonjs') {
              resolved.source = await readFile(new URL(url));
            }
            return resolved;
          }
        }{'}`'});'
```